vsintegration/src/FSharp.Editor/CommonHelpers.fs

```diff
@@ -0,0 +1,216 @@
+                        newData
+                    
+                lexState <- lineData.LexStateAtEndOfLine
+
+                if startLine <= i then
+                    result.AddRange(lineData.ClassifiedSpans |> Seq.filter(fun token ->
+                        textSpan.Contains(token.TextSpan.Start) ||
+                        textSpan.Contains(token.TextSpan.End - 1) ||
+                        (token.TextSpan.Start <= textSpan.Start && textSpan.End <= token.TextSpan.End)))
+
+            // If necessary, invalidate all subsequent lines after endLine
+            if endLine < lines.Count - 1 then 
+                match sourceTextData.[endLine+1] with 
+                | Some data  -> 
+                    if data.LexStateAtStartOfLine <> lexState then
+                        sourceTextData.ClearFrom (endLine+1)
+                | None -> ()
+
+            result
+        with ex -> 
+            Assert.Exception(ex)
+            reraise()  
+
+    let tryClassifyAtPosition (documentKey, sourceText: SourceText, filePath, defines, position: int, cancellationToken) =
+        let textLine = sourceText.Lines.GetLineFromPosition(position)
+        let textLinePos = sourceText.Lines.GetLinePosition(position)
+        let textLineColumn = textLinePos.Character
+
+        let classifiedSpanOption =
+            getColorizationData(documentKey, sourceText, textLine.Span, Some(filePath), defines, cancellationToken)
+            |> Seq.tryFind(fun classifiedSpan -> classifiedSpan.TextSpan.Contains(position))
+
+        match classifiedSpanOption with
+        | Some(classifiedSpan) ->
+            match classifiedSpan.ClassificationType with
+            | ClassificationTypeNames.ClassName
+            | ClassificationTypeNames.DelegateName
+            | ClassificationTypeNames.EnumName
+            | ClassificationTypeNames.InterfaceName
+            | ClassificationTypeNames.ModuleName
+            | ClassificationTypeNames.StructName
+            | ClassificationTypeNames.TypeParameterName
+            | ClassificationTypeNames.Identifier -> 
+                match QuickParse.GetCompleteIdentifierIsland true (textLine.ToString()) textLineColumn with
+                | Some (islandIdentifier, islandColumn, isQuoted) -> 
+                    let qualifiers = if isQuoted then [islandIdentifier] else islandIdentifier.Split '.' |> Array.toList
+                    Some (islandColumn, qualifiers, classifiedSpan.TextSpan)
+                | None -> None
+            | ClassificationTypeNames.Operator ->
+                let islandColumn = sourceText.Lines.GetLinePositionSpan(classifiedSpan.TextSpan).End.Character
+                Some (islandColumn, [""], classifiedSpan.TextSpan) 
+            | _ -> None
+        | _ -> None
+
+open System.Reflection
+open Microsoft.VisualStudio.Shell
+open System.Windows.Controls
+// A port of https://github.com/tomasr/viasfora/blob/master/Viasfora/Util/VsColors.cs
```

Comment:
If you guys are supporting VS 2010 in this release then you'll need to interact with colors this way. In which case I'd add a comment that says something like "We want to support Visual Studio 2010, but EnvironmentColors weren't added until Visual Studio 2012 so we have to load the VS2012 DLL dynamically and interact with themes using reflection."

If this .vsix isn't going to be used in VS2010 then I believe you can take a dependency on `Microsoft.VisualStudio.Shell.11.0` and refer to `EnvironmentColors` directly.